File: web/src/api/cronJob.ts
import request, { type ApiResponse } from './request'

export interface CronJob {
  id: number
  tenant_id: number
  name: string
  cron_expr: string
  command: string
  description: string
  status: number
  last_run_at: string | null
  next_run_at: string | null
  created_at: string
  updated_at: string
}

export interface CronJobLog {
  id: number
  job_id: number
  tenant_id: number
  status: string
  output: string
  error: string
  started_at: string
  ended_at: string | null
  created_at: string
}

export interface CronJobListResult {
  list: CronJob[]
  total: number
  page: number
  page_size: number
}

export interface CronJobLogListResult {
  list: CronJobLog[]
  total: number
  page: number
  page_size: number
}

const cronJobApi = {
  list(page = 1, pageSize = 20): Promise<ApiResponse<CronJobListResult>> {
    return request.get('/cron-jobs', { params: { page, page_size: pageSize } })
  },

  getById(id: number): Promise<ApiResponse<CronJob>> {
    return request.get(`/cron-jobs/${id}`)
  },

  create(data: Partial<CronJob>): Promise<ApiResponse<CronJob>> {
    return request.post('/cron-jobs', data)
  },

  update(id: number, data: Partial<CronJob>): Promise<ApiResponse<CronJob>> {
    return request.put(`/cron-jobs/${id}`, data)
  },

  delete(id: number): Promise<ApiResponse<null>> {
    return request.delete(`/cron-jobs/${id}`)
  },

  toggleStatus(id: number): Promise<ApiResponse<null>> {
    return request.put(`/cron-jobs/${id}/toggle`)
  },

  executeOnce(id: number): Promise<ApiResponse<null>> {
    return request.post(`/cron-jobs/${id}/execute`)
  },

  getLogs(id: number, page = 1, pageSize = 20): Promise<ApiResponse<CronJobLogListResult>> {
    return request.get(`/cron-jobs/${id}/logs`, { params: { page, page_size: pageSize } })
  }
}

export default cronJobApi

File: backend/internal/model/cron_job.go
package model

import (
	"time"
)

type CronJob struct {
	ID          int64      `gorm:"primaryKey" json:"id"`
	TenantID    int64      `gorm:"index;default:0" json:"tenant_id"`
	Name        string    `gorm:"size:100;not null" json:"name"`
	CronExpr    string    `gorm:"size:50;not null" json:"cron_expr"`
	Command     string    `gorm:"size:500;not null" json:"command"`
	Description string    `gorm:"size:500" json:"description"`
	Status      int       `gorm:"default:1" json:"status"`
	LastRunAt   *time.Time `json:"last_run_at"`
	NextRunAt   *time.Time `json:"next_run_at"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

func (CronJob) TableName() string {
	return "cron_jobs"
}

type CronJobLog struct {
	ID        int64      `gorm:"primaryKey" json:"id"`
	JobID     int64      `gorm:"index;not null" json:"job_id"`
	TenantID  int64      `gorm:"index;default:0" json:"tenant_id"`
	Status    string    `gorm:"size:20;not null" json:"status"`
	Output    string    `gorm:"type:text" json:"output"`
	Error     string    `gorm:"type:text" json:"error"`
	StartedAt time.Time `json:"started_at"`
	EndedAt   *time.Time `json:"ended_at"`
	CreatedAt time.Time `json:"created_at"`
}

func (CronJobLog) TableName() string {
	return "cron_job_logs"
}
